app/models/sensor.rb

```diff
@@ -2,6 +2,13 @@ class Sensor < ActiveRecord::Base
   has_many :readings, dependent: :destroy
   has_many :measurement_sensors, dependent: :destroy
   has_many :measurements, through: :measurement_sensors
+  has_many :locations, dependent: :destroy
 
-  validates :long, :lat, presence: true
+  before_create :generate_token
+
+  private
+
+  def generate_token
+    self.token = SecureRandom.urlsafe_base64.gsub('-_','')[0,12]
```

Comment:
Unfortunately it is [hardcoded](https://github.com/robertomiranda/has_secure_token/blob/master/lib/has_secure_token.rb#L35) right now :(